scalyr_agent/scalyr_client.py

```diff
@@ -1585,8 +1585,14 @@ def __set_attributes(self, thread_id, attributes):
                 tmp_buffer.write(b", ")
         if attributes_to_serialize:
             tmp_buffer.write(b"attrs:")
+            # NOTE: attributes is always a dict so we can simply rewinding logic by assuming }
+            # is always the last character when serializing a dict to json
+            tmp_buffer.write(scalyr_util.json_encode(attributes, binary=True)[:-1])
```

Comment:
Hmm... yeah, the purist in me wants to verify that that character is a `}` but I imagine doing that check would make this code less readable and add in overhead in a place we don't want it.  You are right though that this should be very deterministic behavior for the JSON libraries.  If it's not the case, we should get test failures.

Maybe one thing we can do to make it more explicit is have a test case that verifies the last character for a serialized dict is always a '}'.  We have tests that verify common behavior across all JSON libraries.  We'd just add the test there commenting why it is important to us.

What do you think?  Overkill?

Reply:
Yep, I agree about the test case. Will add it for just in case to be on the safe side.

Reply:
I added a test case and verified that the output is the same for both implementations.

Smoke / end to end tests also seem to pass, but we should still keep an eye on it once it's merged.

I think if there are no issues after another load test with this change included, we should be fine :)